rust/relay/src/server.rs

```diff
@@ -955,6 +932,57 @@ where
 
         tracing::info!(target: "relay", channel = %chan.value(), %client, %peer, %allocation, "Channel binding is now deleted (and can be rebound)");
     }
+
+    fn make_debug_error_response(
+        &mut self,
+        error_code: impl Into<ErrorCode>,
+        request: &impl StunRequest,
+    ) -> Message<Attribute> {
+        let error_code = error_code.into();
+
+        tracing::debug!(target: "relay", "{} failed: {}", request.method(), error_code.reason_phrase());
+
+        self.make_error_response(error_code, request)
+    }
+
+    fn make_warn_error_response(
+        &mut self,
+        error_code: impl Into<ErrorCode>,
+        request: &impl StunRequest,
+    ) -> Message<Attribute> {
+        let error_code = error_code.into();
+
+        tracing::warn!(target: "relay", "{} failed: {}", request.method(), error_code.reason_phrase());
+
+        self.make_error_response(error_code, request)
+    }
```

Comment:
Unfortuantely, `tracing` doesn't support passing a dynamic `Level`, otherwise these functions could be avoided and we could instead pass a `Level` as a parameter.

Reply:
hmm, this looks a bit weird though, could we use our own enum we match and log depending on that?

We could event use `Level` and match on that.

Reply:
> We could event use `Level` and match on that.

I had that at some point but it is odd because we only ever use `warn` and `debug`!

You are welcome to have a crack at it, I ended up with this being the least-weird solution :)

Reply:
I will push a commit, let me know what you think, feel free to revert it